apis/v1alpha1/httproute_types.go

```diff
@@ -207,32 +222,67 @@ type HTTPRouteMatch struct {
 	ExtensionRef *RouteMatchExtensionObjectReference `json:"extensionRef" protobuf:"bytes,5,opt,name=extensionRef"`
 }
 
-// HTTPRouteFilter defines a filter-like action to be applied to
-// requests.
-type HTTPRouteFilter struct {
-	// Headers related filters.
-	//
-	// Support: extended
+const (
+	// FilterHTTPReqeustHeader can be used to add or remove an HTTP
+	// header from an HTTP request before it is sent to the upstream target.
+	// Support: core
 	// +optional
-	Headers *HTTPHeaderFilter `json:"headers" protobuf:"bytes,1,opt,name=headers"`
+	FilterHTTPReqeustHeader = "RequestHeader"
+	// TODO(hbagdi): add FilterHTTPRequestMirroring filter
```

Comment:
Is this TODO intended for the current PR?

Reply:
I added this because we decided to add in traffic mirroring as a filter and this would help us close #185. I want to solve this TODO once this PR is merged in. I left it out to speed up the review process.